app/models/mixins/authentication_mixin.rb

```diff
@@ -79,14 +79,7 @@ def authentication_service_account(type = nil)
   end
 
   def required_credential_fields(type)
-    case type.to_s
-    when "bearer"
-      [:auth_key]
-    when "hawkular"
-      []
-    else
-      [:userid]
-    end
+    []
```

Comment:
@yaacov why the previous code was added here in #7950? Why don't we need it anymore?

Is the test you're adding demonstrating that we don't need this in any case? (I suppose there was a reasoning to add this in #7950)